src/games/strategy/engine/data/Unit.java

```diff
@@ -12,16 +15,30 @@
   private int m_hits = 0;
   private final UnitType m_type;
 
+
+  public static Unit createUnit(final UnitType unitType) {
+    return createUnit(unitType, null);
+  }
+
+  public static Unit createUnit(final UnitType unitType, final PlayerID player) {
+    return unitType.create(player);
+  }
+
+  public static Unit createUnit(final String typeName, final GameData data) {
+    return createUnit(typeName, data, null);
+  }
+
+  public static Unit createUnit(final String typeName, final GameData data, final PlayerID owner) {
+    return createUnit( (new UnitType(typeName, data)), owner);
+  }
+
```

Comment:
Some of these createUnit methods aren't used so should be removed. Naming wise its probably better to just name them 'create' rather than 'createUnit'.